### PR TITLE
[EOC-30] create new org form store to db

### PR DESCRIPTION
**Pull request by adamklepacz**

### CODE REVIEW

File: client/src/app/components/Toolbar/Toolbar.jsx [hunk 1]
```diff
@@ -16,28 +16,42 @@ import UserBar from './components/UserBar';
...
 class Toolbar extends PureComponent {
   state = {
+    cohortFormVisbility: false,
     shoppingFormVisbility: false
   };
 
-  handleShoppingFormVisibilty = () => {
-    const { shoppingFormVisbility } = this.state;
-    this.setState({ shoppingFormVisbility: !shoppingFormVisbility });
+  handleFormVisibilty = isShoppingList => {
```
Comment (marekrozmus) [inline client/src/app/components/Toolbar/Toolbar.jsx]: should we check if this is list or cohort? Maybe we should show only one form - the one that was just clicked and hide all others? Clicking outside of form (if it is empty) should cliese the form popup.

File: client/src/modules/cohort/model/actions.js [hunk 2]
```diff
@@ -1,9 +1,28 @@
...
+
+const fetchCohortsSuccess = data => ({
+  type: CohortActionTypes.FETCH_COHORTS_SUCCESS,
+  payload: data
 });
 
 // Dispatchers
-export const fetchCohorts = () => dispatch => dispatch(fetchCohortsSuccess());
+export const createCohort = (name, description) => dispatch =>
+  postData(`${ENDPOINT_URL}/cohorts/new-cohort`, { name, description })
```
Comment (marekrozmus) [inline client/src/modules/cohort/model/actions.js]: just `new` is enough - we got info about what will be created before /

File: client/src/modules/shopping-list/components/CreationForm.jsx [hunk 3]
```diff
@@ -57,6 +57,7 @@ class CreationForm extends Component {
 }
 
 CreationForm.propTypes = {
+  isShoppingList: PropTypes.bool,
```
Comment (marekrozmus) [inline client/src/modules/shopping-list/components/CreationForm.jsx]: remove this - the form should not know anything about shopping lists or cohorts - it should be also common component

File: server/models/cohort.model.js [hunk 4]
```diff
@@ -0,0 +1,16 @@
+const mongoose = require('mongoose');
+
+const { Schema } = mongoose;
+
+const CohortSchema = new Schema(
+  {
+    adminIds: { type: String },
+    description: { type: String },
```
Comment (marekrozmus) [inline server/models/cohort.model.js]: no field is required?

File: server/routes/cohort.js [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+
+const {
+  createCohort,
+  getCohortById,
+  getCohorts
+} = require('../controllers/cohort');
+
+router.get('/', getCohorts);
+router.get('/:id', getCohortById);
+router.post('/new-cohort', createCohort);
```
Comment (marekrozmus) [inline server/routes/cohort.js]: remove `-cohort` from route

File: server/controllers/cohort.js [hunk 6]
```diff
@@ -0,0 +1,41 @@
+const Cohort = require('../models/cohort.model');
+
+const createCohort = (req, resp) => {
+  const { description, name } = req.body;
+
+  const cohort = new Cohort({
+    name,
+    description
+  });
```
Comment (alemikolo) [inline server/controllers/cohort.js]: Shouldn't we save user id here (as adminId and memberId)?

Comment (adamklepacz) [reply server/controllers/cohort.js]: on creation we are saving ownerId and adminId

File: client/src/modules/cohort/model/actions.js [hunk 7]
```diff
@@ -14,12 +14,11 @@ const fetchCohortsSuccess = data => ({
...
 
 // Dispatchers
-export const createCohort = (name, description, ownerId) => dispatch =>
+export const createCohort = (name, description, adminId) => dispatch =>
   postData(`${ENDPOINT_URL}/cohorts/create`, {
     name,
     description,
-    ownerId,
-    adminIds: ownerId
+    adminIds: adminId
```
Comment (alemikolo) [inline client/src/modules/cohort/model/actions.js]: shouldn't adminIds be an array? See in shoppingList controller also. Here string is saved... (Cohorts also...)

File: client/src/modules/cohort/model/actions.js [hunk 8]
```diff
@@ -1,9 +1,32 @@
...
+
+const fetchCohortsSuccess = data => ({
+  type: CohortActionTypes.FETCH_COHORTS_SUCCESS,
+  payload: data
 });
 
 // Dispatchers
-export const fetchCohorts = () => dispatch => dispatch(fetchCohortsSuccess());
+export const createCohort = (name, description, adminId) => dispatch =>
+  postData(`${ENDPOINT_URL}/cohorts/create`, {
```
Comment (marekrozmus) [inline client/src/modules/cohort/model/actions.js]: frondend does not need to know db structure - it should only send data and server will take care of use that data to create correct db entry.

File: client/src/modules/shopping-list/model/actions.js [hunk 9]
```diff
@@ -27,8 +27,12 @@ export const fetchProducts = () => dispatch =>
     .then(json => dispatch(recieveProducts(json)))
     .catch(err => dispatch(fetchProductsError(err)));
 
-export const createNewShoppingList = (name, description) => dispatch =>
-  postData(`${ENDPOINT_URL}/shopping-lists/new-list`, { name, description })
+export const createShoppingList = (name, description, adminId) => dispatch =>
+  postData(`${ENDPOINT_URL}/shopping-lists/create`, {
+    name,
+    description,
+    adminIds: adminId
```
Comment (marekrozmus) [inline client/src/modules/shopping-list/model/actions.js]: same here about data structure

File: server/controllers/cohort.js [hunk 10]
```diff
@@ -0,0 +1,42 @@
+const Cohort = require('../models/cohort.model');
+
+const createCohort = (req, resp) => {
+  const { description, name, adminIds } = req.body;
```
Comment (marekrozmus) [inline server/controllers/cohort.js]: only one creator id should be send from front end

File: client/src/modules/cohort/model/actions.js (hunk 7 above)
Comment (adamklepacz) [reply client/src/modules/cohort/model/actions.js]: it is saved as array, it's controlled by model

Comment (adamklepacz) [reply client/src/modules/cohort/model/actions.js]: zrzut ekranu 2019-02-04 o 12 20 13

**Pull request merged**